src/standalone/standard.rs

```diff
@@ -137,34 +123,68 @@ impl StandardDocumentOps for StandaloneStandardClient {
 
         Ok(encrypted_doc)
     }
+
     async fn decrypt(
         &self,
         encrypted_document: EncryptedDocument,
         metadata: &AlloyMetadata,
     ) -> Result<PlaintextDocument, AlloyError> {
-        let dek =
-            Self::decrypt_document_dek(encrypted_document.edek, &self.config.secrets, metadata)?;
+        let (_, dek) = Self::decrypt_document_dek(
+            encrypted_document.edek,
+            &self.config.secrets,
+            &metadata.tenant_id,
+        )?;
         decrypt_document_core(encrypted_document.document, dek)
     }
 
-    fn get_searchable_edek_prefix(&self, id: i32) -> Vec<u8> {
-        get_prefix_bytes_for_search(ironcore_documents::key_id_header::KeyIdHeader::new(
-            Self::get_edek_type(),
-            Self::get_payload_type(),
-            KeyId(id as u32),
-        ))
-        .into()
+    async fn rekey_edeks(
+        &self,
+        edeks: HashMap<String, EdekWithKeyIdHeader>,
+        metadata: &AlloyMetadata,
+        new_tenant_id: Option<TenantId>,
+    ) -> Result<RekeyEdeksBatchResult, AlloyError> {
+        let parsed_new_tenant_id = new_tenant_id.as_ref().unwrap_or(&metadata.tenant_id);
+        let rekey_edek = |edek: EdekWithKeyIdHeader| {
+            let (key_id, dek) =
+                Self::decrypt_document_dek(edek, &self.config.secrets, &metadata.tenant_id)?;
+            let secret = self.config.secrets.get(&key_id.0).ok_or_else(|| {
+                AlloyError::InvalidConfiguration(format!(
+                    "Provided secret id `{}` does not exist in the standard configuration.",
+                    &key_id.0
+                ))
+            })?;
+            let encryption_key = derive_aes_encryption_key(&secret.secret, parsed_new_tenant_id);
+            let (dek, v4_doc) = generate_aes_edek_and_sign(
+                &mut *get_rng(&self.rng),
+                encryption_key,
+                Some(dek), // Same dek will be returned
+                key_id.0.to_string().as_str(),
+            )?;
```

Comment:
it's because it was the same that it felt okay to shadow. But because of a different comment, the `dek` is no longer needed again, so it's no longer shadowed here.